### PR TITLE
#14719  url params parsing should not be lenient

### PR DESCRIPTION
**Pull request by martinstuga**
There is my initial commit with a generic and centralized parameter validation.

This validation is being done just for GET requests. Validation on the other methods will have to be done en every single rest request.

### CODE REVIEW

**Comment by martinstuga** [comment]
I've signed the CLA, don't know why GitHub is complaining.

Anybody can help me?

File: core/src/main/java/org/elasticsearch/search/aggregations/metrics/stats/StatsAggregator.java [hunk 1]
```diff
@@ -43,7 +43,7 @@
 /**
  *
  */
-public class StatsAggegator extends NumericMetricsAggregator.MultiValue {
+public class StatsAggregator extends NumericMetricsAggregator.MultiValue {
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/search/aggregations/metrics/stats/StatsAggregator.java]: This change is not related to the remainder of this pull request; can you please remove it from this change set?

Comment (martinstuga) [inline core/src/main/java/org/elasticsearch/search/aggregations/metrics/stats/StatsAggregator.java]: You're right. Excuse me!
